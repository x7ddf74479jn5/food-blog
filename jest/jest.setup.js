// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
