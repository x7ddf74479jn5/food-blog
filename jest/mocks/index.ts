export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
