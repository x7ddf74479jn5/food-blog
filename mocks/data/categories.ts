export const categories = {
  rice: { id: "1", slug: "rice", name: "ご飯物", color: "grey" },
  salad: { id: "2", slug: "salad", name: "サラダ", color: "grey" },
  tips: { id: "3", slug: "tips", name: "料理基礎", color: "grey" },
};
