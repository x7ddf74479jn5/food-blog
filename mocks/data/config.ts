import type { TConfig } from "@/types/models/config";

import { dateCommon } from "./utils";

export const mockConfig: TConfig = {
  ...dateCommon,
  apiHost: "https://food-blog.microcms.io/api/v1/",
  host: "https://food-blog-chi.vercel.app/",
  siteTitle: "Food Blog",
  siteDescription: "料理レシピを紹介するブログ",
  siteKeywords: ["料理"],
  siteImage: { url: "https://food-blog-chi.vercel.app/site_image.jpg", width: 100, height: 100 },
  organization: "Pandashark",
  officialSite: "https://next-portfolio-livid.vercel.app/",
  twitterId: "@pandashark6",
  email: "x7ddf74479jn5@gmail.com",
};
