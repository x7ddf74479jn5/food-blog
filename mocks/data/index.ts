export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
