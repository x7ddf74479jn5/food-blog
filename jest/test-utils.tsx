import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
