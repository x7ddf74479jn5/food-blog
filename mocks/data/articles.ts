import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
