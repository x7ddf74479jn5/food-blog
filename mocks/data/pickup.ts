import { mockArticles } from "@mocks/data";
import { dateCommon } from "@mocks/data/utils";

export const mockPickup = {
  id: "1",
  name: "旬の食材を使ったおすすめレシピ",
  description: "旬の食材を使ったおすすめレシピ",
  startDate: new Date("2021/10/08 00:00"),
  endDate: new Date("2023/02/08 00:00"),
  articles: Object.values(mockArticles),
  ...dateCommon,
};
