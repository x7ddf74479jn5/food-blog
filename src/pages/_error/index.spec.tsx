import { mockCategories, mockConfig } from "@mocks/data";
import { render, screen } from "jest/test-utils";
import renderer from "react-test-renderer";

import ErrorPage from "./index.page";

describe("pages/_error", () => {
  const mockCategoryList = Object.values(mockCategories);
  it("snapshot", () => {
    const tree = renderer.create(<ErrorPage config={mockConfig} categories={mockCategoryList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("OK: 初期レンダリング", async () => {
    const { unmount } = render(<ErrorPage config={mockConfig} categories={mockCategoryList} />);
    const h1 = screen.getByRole("heading", { level: 1 });
    expect(h1).toHaveTextContent("Unhandled Error");
    unmount();
  });
});
