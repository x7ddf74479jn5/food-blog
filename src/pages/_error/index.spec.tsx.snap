// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/_error snapshot 1`] = `
<div
  className="flex flex-col px-4 mx-auto mb-16 max-w-screen-lg min-h-screen"
>
  <header
    className="py-2"
  >
    <div
      className="flex justify-between items-center"
    >
      <a
        className="text-2xl font-bold text-green-600"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Food Blog
      </a>
      <div
        className="hidden sm:block w-1/3"
      >
        <div
          className="relative"
        >
          <div
            className="flex items-center py-2 px-3 space-x-2 bg-gray-100 dark:bg-gray-700 rounded-full"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "gray",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              className="w-full dark:text-gray-100 bg-gray-100 dark:bg-gray-700 focus:outline-none"
              inputMode="text"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              onSubmit={[Function]}
              placeholder="Search..."
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="flex items-center space-x-4"
      >
        <div
          style={
            Object {
              "MozTransition": "opacity 0.25s",
              "MozUserSelect": "none",
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "WebkitTransition": "opacity 0.25s",
              "WebkitUserSelect": "none",
              "borderRadius": 12,
              "direction": "ltr",
              "display": "inline-block",
              "msUserSelect": "none",
              "opacity": 1,
              "position": "relative",
              "textAlign": "left",
              "touchAction": "none",
              "transition": "opacity 0.25s",
              "userSelect": "none",
            }
          }
        >
          <div
            className="react-switch-bg"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "MozTransition": "background 0.25s",
                "WebkitTransition": "background 0.25s",
                "background": "#555555",
                "borderRadius": 12,
                "cursor": "pointer",
                "height": 24,
                "margin": 0,
                "position": "relative",
                "transition": "background 0.25s",
                "width": 48,
              }
            }
          >
            <div
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "WebkitTransition": "opacity 0.25s",
                  "height": 24,
                  "opacity": 0,
                  "pointerEvents": "none",
                  "position": "relative",
                  "transition": "opacity 0.25s",
                  "width": 27,
                }
              }
            >
              <div
                className="flex justify-center items-center h-full"
              >
                <svg
                  fill="currentColor"
                  height="80%"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "yellow",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="80%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                  />
                </svg>
              </div>
            </div>
            <div
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "WebkitTransition": "opacity 0.25s",
                  "height": 24,
                  "opacity": 1,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transition": "opacity 0.25s",
                  "width": 27,
                }
              }
            >
              <div
                className="flex justify-center items-center h-full"
              >
                <svg
                  fill="currentColor"
                  height="80%"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "gold",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="80%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="react-switch-handle"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                "background": "#ffffff",
                "border": 0,
                "borderRadius": "50%",
                "boxShadow": null,
                "cursor": "pointer",
                "display": "inline-block",
                "height": 20,
                "outline": 0,
                "position": "absolute",
                "top": 2,
                "transform": "translateX(2px)",
                "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                "width": 20,
              }
            }
          />
          <input
            aria-checked={false}
            aria-label="switch between day and night themes"
            checked={false}
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            role="switch"
            style={
              Object {
                "border": 0,
                "clip": "rect(0 0 0 0)",
                "height": 1,
                "margin": -1,
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "width": 1,
              }
            }
            type="checkbox"
          />
        </div>
        <div
          className="inline-block relative text-left"
        >
          <span
            className="rounded-md shadow-sm"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="inline-flex justify-center py-1 px-3 w-full text-sm font-medium leading-5 text-gray-700 hover:text-gray-500 dark:text-gray-100 bg-white dark:bg-gray-700 rounded-md border border-gray-300 dark:border-gray-500 dark:hover:border-gray-400 transition duration-150 ease-in-out focus:outline-none"
              id="headlessui-menu-button-1"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span>
                カテゴリー
              </span>
              <svg
                className="-mr-1 ml-2 w-5 h-5"
                fill="currentColor"
                viewBox="0 0 20 20"
              >
                <path
                  clipRule="evenodd"
                  d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      className="block sm:hidden mt-2"
    >
      <div
        className="relative"
      >
        <div
          className="flex items-center py-2 px-3 space-x-2 bg-gray-100 dark:bg-gray-700 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "gray",
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            className="w-full dark:text-gray-100 bg-gray-100 dark:bg-gray-700 focus:outline-none"
            inputMode="text"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            onSubmit={[Function]}
            placeholder="Search..."
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="flex flex-col gap-12 justify-center items-center mt-8"
    role="alert"
  >
    <svg
      className="w-32 h-32 text-gray-500"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 496 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm129.1-303.8c-3.8-4.4-10.3-5.4-15.3-2.5l-80 48c-3.6 2.2-5.8 6.1-5.8 10.3s2.2 8.1 5.8 10.3l80 48c5.4 3.2 11.8 1.6 15.3-2.5 3.8-4.5 3.9-11 .1-15.5L343.6 208l33.6-40.3c3.8-4.5 3.7-11.1-.1-15.5zM220 208c0-4.2-2.2-8.1-5.8-10.3l-80-48c-5-3-11.5-1.9-15.3 2.5-3.8 4.5-3.9 11-.1 15.5l33.6 40.3-33.6 40.3c-3.8 4.5-3.7 11 .1 15.5 3.5 4.1 9.9 5.7 15.3 2.5l80-48c3.6-2.2 5.8-6.1 5.8-10.3zm28 64c-45.4 0-100.9 38.3-107.8 93.3-1.5 11.8 6.9 21.6 15.5 17.9C178.4 373.5 212 368 248 368s69.6 5.5 92.3 15.2c8.5 3.7 17-6 15.5-17.9-6.9-55-62.4-93.3-107.8-93.3z"
      />
    </svg>
    <div
      className="flex flex-col gap-8 mx-auto text-center"
    >
      <h1
        className="flex justify-center text-4xl font-bold"
      >
        Unhandled Error
      </h1>
      <p>
        サイト上で問題が発生しました。
      </p>
    </div>
    <button
      className="flex flex-row gap-2 justify-center items-center py-2 px-4 text-white bg-green-700 hover:bg-opacity-90 rounded-xl border-2 border-green-600"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        />
      </svg>
      <span>
        Homeに戻る
      </span>
    </button>
  </div>
</div>
`;
